exchange-v2/contracts/ExchangeV2Core.sol

```diff
@@ -23,6 +23,9 @@ abstract contract ExchangeV2Core is Initializable, OwnableUpgradeable, AssetMatc
     //on-chain orders
     mapping(bytes32 => OrderAndFee) public onChainOrders;
 
+    //prevent reentrancy
+    bool internal locked;
```

Comment:
can we fix reentrancy without adding new variable? for example setting internal variables before transfer

Reply:
and in "@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol" this variable is initialized, but here it's not. is it ok? won't it break something?

Reply:
done, delete modifier